### PR TITLE
Adding Factors API

### PR DESCRIPTION
**Pull request by robertjd**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,5 @@ node_modules
 jsdocs/
 jsdoc/
 
+.envrc
```
Comment (lboyette-okta) [inline .gitignore]: what generated this?

File: package.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
     "ink-docstrap": "^1.3.0",
     "jsdoc": "^3.4.0",
     "mocha": "^3.4.1",
-    "nyc": "^10.3.2"
+    "nyc": "^10.3.2",
+    "speakeasy": "^2.0.0"
```
Comment (lboyette-okta) [inline package.json]: speakeasy?

edit: cool, testing totp

File: test/it/factor-verify.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      credentials: {
+        password: { value: 'Abcd1234' }
+      }
+    };
+    // Cleanup the user if user exists
+    await utils.cleanup(client, newUser);
+    createdUser = await client.createUser(newUser);
+  });
+
+  it('should allow me to verify an Security Question factor', async () => {
```
Comment (lboyette-okta) [inline test/it/factor-verify.js]: nit: `verify an Security` => `verify a Security`

File: test/it/factor-activate.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const speakeasy = require('speakeasy');
+
+const utils = require('../utils');
+const okta = require('../../');
+const expect = require('chai').expect;
+let orgUrl = process.env.OKTA_CLIENT_ORGURL;
+
+if (process.env.OKTA_USE_MOCK) {
+  orgUrl = `${orgUrl}/group-get`;
```
Comment (lboyette-okta) [inline test/it/factor-activate.js]: probably want to rename this to `factor-activate`

File: test/it/factor-create.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
+const utils = require('../utils');
+const okta = require('../../');
+const models = require('../../src/models');
+const expect = require('chai').expect;
+let orgUrl = process.env.OKTA_CLIENT_ORGURL;
+
+if (process.env.OKTA_USE_MOCK) {
+  orgUrl = `${orgUrl}/group-get`;
```
Comment (lboyette-okta) [inline test/it/factor-create.js]: ah, rename here and for the other tests too

File: .gitignore (hunk 1 above)
Comment (robertjd) [reply .gitignore]: https://direnv.net/

File: package.json (hunk 2 above)
Comment (robertjd) [reply package.json]: super cool!

**Pull request merged**